src/game/common/system/geometry.cpp

```diff
@@ -248,10 +248,19 @@ float GeometryInfo::Is_Intersected_By_Line_Segment(Coord3D &loc, Coord3D &from,
 void GeometryInfo::Calc_Pitches(
     const Coord3D &coord1, const GeometryInfo &info, const Coord3D &coord2, float &f1, float &f2) const
 {
-    float z = Get_ZDelta_To_Center_Position() + coord1.z;
-    float xy = Sqrt(Square(coord2.y - coord1.y) + Square(coord2.x - coord1.x));
-    f2 = Atan2(xy, float(coord2.z + info.Get_Max_Height_Above_Position()) - z);
-    f1 = Atan2(xy, float(coord2.z - info.Get_Max_Height_Below_Position()) - z);
+    float x;
+    float y;
+
+    Coord3D coord3;
+    Get_Center_Position(coord1, coord3);
+
+    x = Sqrt(Square(coord2.y - coord3.y) + Square(coord2.x - coord3.x));
+
+    y = coord2.z + info.Get_Max_Height_Above_Position() - coord3.z;
+    f2 = Atan2(y, x);
+
+    y = coord2.z - info.Get_Max_Height_Below_Position() - coord3.z;
```

Comment:
I feel this function would make more sense if on the first line we did:
```cpp
const auto coord_diff = coord2 - coord3;
```
Then all the sums look much simpler:
```cpp
const auto adj = Sqrt(Square(coord_diff.x) + Square(coord_diff.y));
const auto opp_below = coord_diff.z - info.Get_Max_Height_Below_Position();
const auto opp_above = coord_diff.z - info.Get_Max_Height_Above_Position();
f1 = Atan2(opp_below, adj);
f2 = Atan2(opp_above, adj);